tests/plugins/npm_test.py

```diff
@@ -0,0 +1,22 @@
+import pytest
+
+from detect_secrets.plugins.npm import NpmDetector
+
+
+class TestNpmDetector:
+
+    @pytest.mark.parametrize(
+        'payload, should_flag',
+        [
+            ('//registry.npmjs.org/:_authToken=xxxxxxxxxxxxxxxxxxxx', True),
+            ('//registry.npmjs.org:_authToken=xxxxxxxxxxxxxxxxxxxx', False),
+            ('registry.npmjs.org/:_authToken=xxxxxxxxxxxxxxxxxxxx', False),
+            ('///:_authToken=xxxxxxxxxxxxxxxxxxxx', False),
+            ('_authToken=xxxxxxxxxxxxxxxxxxxx', False),
+            ('foo', False),
+        ],
+    )
+    def test_analyze(self, payload, should_flag):
+        logic = NpmDetector()
+        output = logic.analyze_line(payload, 1, 'mock_filename')
```

Comment:
Hi @ninoseki,

Thanks so much for your PR! Sorry it has been a while since one of our team members has looked into this -- but I'm back now, and will be paying closer attention to this project. We appreciate your contribution to helping make this tool better!

Would you mind merging this into the `pre-v1-launch` branch instead? All you'd have to do is change this line as such:

```suggestion
        output = logic.analyze_line(filename='mock_filename', line=payload)
```

and it will greatly assist our process in revamping the underlying architecture of this tool (by reducing merge conflicts).